internal/olm/operator/registry/configmap/configmap_test.go

```diff
@@ -0,0 +1,277 @@
+		It("should return the registry configmap name", func() {
+			val := "Test"
+			name := k8sutil.FormatOperatorNameDNS1123(val)
+			ans := fmt.Sprintf("%s-registry-manifests", name)
+
+			Expect(getRegistryConfigMapName("Test")).Should(Equal(ans))
+		})
+	})
+
+	Describe("makeObjectFileName", func() {
+		var (
+			fileName  string
+			testVal   []byte
+			userInput []string
+		)
+		BeforeEach(func() {
+			testVal = []byte("test")
+			userInput = []string{"userInput", "userInput2"}
+			fileName = hashContents(testVal) + "."
+		})
+
+		It("returns the filename with extra user given string", func() {
+			for _, name := range userInput {
+				if name != "" {
+					fileName += strings.ToLower(name) + "."
+				}
+			}
+			fileName = fileName + "yaml"
+			Expect(makeObjectFileName(testVal, userInput...)).Should(Equal(fileName))
+		})
+
+		It("returns the filename without extra user given string", func() {
+			fileName = fileName + "yaml"
+			Expect(makeObjectFileName(testVal)).Should(Equal(fileName))
+		})
+	})
+
+	Describe("addObjectToBinaryData ", func() {
+		It("should add given object to the given binaryData", func() {
+			type binaryData map[string][]byte
+			userInput := []string{"userInput", "userInput2"}
+
+			type test_struct struct {
+				val1 string
+				val2 string
+			}
+			b := make(binaryData)
+			// fmt.Printf("%+v", b)
+			obj := test_struct{"val1", "val2"}
+
+			binaryData_local := make(binaryData)
+			b_local, err := yaml.Marshal(obj)
+
+			binaryData_local[makeObjectFileName(b_local, userInput...)] = b_local
+
+			Expect(err).Should(BeNil())
+			Expect(addObjectToBinaryData(b, obj, userInput...)).Should(BeNil())
+			Expect(b).Should(Equal(binaryData_local))
+		})
+
+	})
+
+	Describe("makeObjectBinaryData", func() {
+		It("creates the binary data", func() {
+			binaryData := make(map[string][]byte)
+			type return_struct struct {
+				val1 map[string][]byte
+				val2 error
+			}
+			type test_struct struct {
+				val1 string
+				val2 string
+			}
+
+			obj := test_struct{"val1", "val2"}
+			userInput := []string{"userInput", "userInput2"}
+			b := make(map[string][]byte)
+			b, e := makeObjectBinaryData(obj, userInput...)
+
+			addObjectToBinaryData(binaryData, obj, userInput...)
+
+			Expect(e).Should(BeNil())
+			Expect(b).Should(Equal(binaryData))
+
+		})
+	})
+
+	Describe("makeBundleBinaryData", func() {
+		It("should serialize bundle to binary data", func() {
+			var e error
+			b := apimanifests.Bundle{
+				Name: "testbundle",
+				Objects: []*unstructured.Unstructured{
+					{
+						Object: map[string]interface{}{"val1": "val1"},
+					},
+					{
+						Object: map[string]interface{}{"val2": "va2"},
+					},
+				},
+			}
+
+			binaryData, err := makeBundleBinaryData(&b)
+			val := make(map[string][]byte)
+			for _, obj := range b.Objects {
+				e = addObjectToBinaryData(val, obj, obj.GetName(), obj.GetKind())
+			}
+
+			Expect(binaryData).Should(Equal(val))
+			Expect(e).Should(BeNil())
```

Comment:
This `Expect(e)` should be *IN* the for loop because all you are verifying is if the LAST call to `addObjectToBinaryData` worked.